### PR TITLE
Fix Blazor boot config loading

### LINKED ISSUE #40245: Cannot load blazor in an angular app
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

We have a large angular app and for some parts of this app, we want to use blazor because we have a lot of code in c# and we want to use it in the browser. 
For booting blazor we load the script '_framework/blazor.webassembly.js' with the attribute 'autostart'=false and after loading we call Blazor.start(). This scenario worked before netcore6.0. 
We think the changes in the file ['BootConfig.ts'](https://github.com/dotnet/aspnetcore/blob/0ca2ed9af69e7e334b8e3c1de1d015017f138988/src/Components/Web.JS/src/Platform/BootConfig.ts) commit https://github.com/dotnet/aspnetcore/commit/0ca2ed9af69e7e334b8e3c1de1d015017f138988 breaks this way.
The comparison 'instanceof Promise' doesn't work, because zone.js patches the native promise. 

### Describe the solution you'd like

We know this isn't the standard way to use Blazor, but we cannot write the whole app in blazor so we have to use it in this way. Micro-Frontends are very popular, so it should be possible to use different frameworks for one app. If we patch this comparison everything works fine. Maybe there is common solution for this problem.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by MackinnonBuck**
# Fix Blazor boot config loading

There was an `instanceof Promise` check in `BootConfig.ts` that failed when the default `Promise` type got overridden (as it does in libraries like `zone.js`). This PR fixes the issue by rearranging the logic to avoid performing this check.

The filed issue used Blazor components in an Angular app, where Blazor was manually started using `Blazor.start()` rather than letting the script auto-start. I've reproduced original bug and verified the fix locally.

Fixes #40245

### CODE REVIEW

File: src/Components/Web.JS/src/Platform/BootConfig.ts [hunk 1]
```diff
@@ -14,9 +14,15 @@ export class BootConfigResult {
...
-    const bootConfigResponse = loaderResponse instanceof Promise ?
-      await loaderResponse :
-      await defaultLoadBlazorBootJson(loaderResponse ?? '_framework/blazor.boot.json');
+    let bootConfigResponse: Response;
+
+    if (!loaderResponse) {
+      bootConfigResponse = await defaultLoadBlazorBootJson('_framework/blazor.boot.json');
+    } else if (typeof loaderResponse === 'string') {
+      bootConfigResponse = await defaultLoadBlazorBootJson(loaderResponse);
+    } else {
```
Comment (TanayParikh) [inline src/Components/Web.JS/src/Platform/BootConfig.ts:23]: Is there any further type enforcement we can do here, and then have an `else throw` case at the end?

**Review by TanayParikh — COMMENTED** [review]
Are we planning on backporting this to 6.0?

**Comment by MackinnonBuck** [comment]
> Are we planning on backporting this to 6.0?

Yep, I'll create a backport PR after this is merged.

File: src/Components/Web.JS/src/Platform/BootConfig.ts (hunk 1 above)
Comment (TanayParikh) [reply src/Components/Web.JS/src/Platform/BootConfig.ts:23]: Oh got it, thanks for the clarification. This looks great in that case.

**Review by TanayParikh — APPROVED** [review]

**Pull request merged**